python/pyspark/ml/common.py

```diff
@@ -96,15 +96,15 @@ def _java2py(sc, r, encoding="bytes"):
             return RDD(jrdd, sc)
 
         if clsName == "Dataset":
-            return DataFrame(r, SQLContext.getOrCreate(sc))
+            return DataFrame(r, SparkSession(sc))
```

Comment:
Hm .. I think it might need a bigger scope of change.
<img width="759" alt="Screen Shot 2021-11-16 at 9 30 24 AM" src="https://user-images.githubusercontent.com/6477701/141874067-2273d1c5-a212-4f69-9e0c-dc55258b95c3.png">

I think the constructor currently expects SQLContext ..

Reply:
Hmm, I didn't look too deeply since my changed worked locally and passed tests. But I'll look into this.

Shall I also [update MLlib](https://github.com/apache/spark/pull/34606#issuecomment-969144498)?

Reply:
I think we can update together .. I suspect that would be easier actually.